### PR TITLE
Remove  deprecated CSS tooling. Introduce postcss.

### LINKED ISSUE #1108: Generate `./bin/dev` script
When I generate a new Rails 7 app from suspenders, I want to be able to run `./bin/dev ` in the generated app so that my developer experience is consistent with a vanilla Rails 7 app.

### LINKED ISSUE #1111: Use correct script tags
## Steps to reproduce

1. Checkout the [support_rails_7](https://github.com/thoughtbot/suspenders/tree/support_rails_7) branch.
2. Once you are in the `suspenders` directory, run `./bin/suspenders some_app_name` to generate a Rails app from the current state of suspenders.
3. Change directories into `some_app_name`.
4. Run `rails g scaffold some_model some_attribute` followed by `rails db:migrate`.
5. Run `rails s` and visit a route for the resource created in step 4.

CleanShot 2022-12-14 at 10 00 19@2x

### LINKED ISSUE #1112: Simplify asset dependencies
- [x] Remove bourbon
- [x] Remove normalize.css
- [x] Remove sprockets
- [x] Make cssbunding-rails and postcss the deafults

### PR DESCRIPTION
**Pull request by stevepolitodesign**
## 🚧 Work in progress

- Closes #1112
- Closes #1111
- Closes #1108

## ✅ To-do

- [x] Remove sprockets except for API mode
- [x] Remove obsolete tests
- [x] Add test coverage for the new `css` option
- [x] Add test coverage for the Gems
- [x] #1111 
- [x] Manually test that the generated app loads _new_ CSS and JS

### CODE REVIEW

**Comment by stevepolitodesign** [comment]
We might also want to explore removing the following Gems from `templates/Gemfile.erb` and seeing if the build command adds those gems automatically.

```ruby
gem "jsbundling-rails"
gem "cssbundling-rails"
```

File: lib/suspenders.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
 require "suspenders/generators/advisories_generator"
 require "suspenders/generators/app_generator"
 require "suspenders/generators/static_generator"
-require "suspenders/generators/stylesheet_base_generator"
+# require "suspenders/generators/stylesheet_base_generator"
```
Comment (stevepolitodesign) [inline lib/suspenders.rb:7]: We'll want to delete this module and remove this line.

File: lib/suspenders/generators/app_generator.rb [hunk 2]
```diff
@@ -28,8 +28,11 @@ class AppGenerator < Rails::Generators::AppGenerator
     class_option :skip_test, type: :boolean, default: true,
       desc: "Skip Test Unit"
 
-    class_option :skip_system_test,
-      type: :boolean, default: true, desc: "Skip system test files"
+    class_option :css,
```
Comment (stevepolitodesign) [inline lib/suspenders/generators/app_generator.rb:31]: This is the spirit of this pull-request, so we'll probably want a test for this.

This makes it so a new Rails app is generated via `--css=postcss`.

**Comment by DoodlingDev** [comment]
something to consider tomorrow (today? when are you reading this anyway?)

deleting template/descriptions/stylesheet_base.md
bourbon in templates/application.scss

the tests that are failing because they expect to see bourbon in the gemfile.. I think we can just get rid of those, I thought briefly about reversing their polarity to test that bourbon and bitters _aren't_ installed but I talked myself out of it.

File: templates/suspenders_layout.html.erb.erb [hunk 3]
```diff
@@ -1,21 +1,23 @@
...
   %>
   <title><%%= title %></title>
-  <%%= stylesheet_link_tag :application, media: "all" %>
+  <meta charset="utf-8" />
+  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
   <%%= csrf_meta_tags %>
+  <%%= csp_meta_tag %>
+
+  <%%= stylesheet_link_tag "application", "data-turbo-track": "reload" %>
+  <%%= javascript_include_tag "application", "data-turbo-track": "reload", defer: true %>
```
Comment (stevepolitodesign) [inline templates/suspenders_layout.html.erb.erb:16]: I generated a new Rails app without suspenders, and this is what was in `application.html.erb`.

```sh
rails g new_app -c=postcss
```

File: templates/Gemfile.erb [hunk 4]
```diff
@@ -28,7 +25,8 @@ gem "tzinfo-data", platforms: [:mingw, :x64_mingw, :mswin, :jruby]
 <%# gem "webpacker" %>
```
Comment (stevepolitodesign) [inline templates/Gemfile.erb:25]: I should just remove this now.

File: spec/features/new_project_spec.rb [hunk 5]
```diff
@@ -270,22 +263,43 @@ def app_name
...
     gemfile = read_project_file("Gemfile")
 
-    expect(gemfile).to match(/bourbon/)
+    expect(gemfile).to match(/cssbundling-rails/)
+    expect(gemfile).to match(/jsbundling-rails/)
+    expect(File).to exist("#{project_path}/postcss.config.js")
+    expect(File).to exist("#{project_path}/package.json")
+    expect(File).to exist("#{project_path}/bin/dev")
+    expect(File).to exist("#{project_path}/app/assets/stylesheets/application.postcss.css")
+    expect(File).to exist("#{project_path}/app/javascript/application.js")
```
Comment (stevepolitodesign) [inline spec/features/new_project_spec.rb:280]: This ensures the `--css=postcss` flag did it's job.

**Pull request merged**